Recruit-Training/Embedded-Recruit-Training/Core/Src/GreenLedToggleTask.c

```diff
@@ -1,16 +1,48 @@
 #include "GreenLedToggleTask.h"
+static const GREEN_LED_STATUS_STDID = 0xDDD;
+const TEN_HZ_TO_MS = 10000;
 
 void greenLedToggleTask(void const* arg)
 {
+    static const GREEN_LED_TOGGLE_FREQ = TEN_HZ_TO_MS;
     //One time osDelayUntil initialization
     uint32_t prevWakeTime = osKernelSysTick();
 
     osMutexId_t* canMutex = (osMutexId_t*)arg;
 
     for (;;)
     {
-        osDelayUntil(prevWakeTime + 1);  //TODO: Replace 1 with time period for delay
+        osDelayUntil(prevWakeTime + GREEN_LED_TOGGLE_FREQ);  //TODO: Replace 1 with time period for delay
         //TODO: Check blue toggel flag and toggle blue LED
+        if(greenFlag == 1)
+        {
+            //toggle green led, this requires a HAL GPIO Function
+            HAL_GPIO_TogglePin(GPIOA, GPIO_PIN_6);
+        }
+        GPIO_PinState pinStatus = HAL_GPIO_ReadPin(GPIOA, GPIO_PIN_6);
         //TODO: Send CAN message indicating current state of LED
+        osStatus_t status = osMutexAcquire(canMutex, 1);
+
+        while(status != osOK){
+            status = osMutexAcquire(canMutex, 1);
+        }
+
+        HAL_CAN_GetTxMailboxesFreeLevel(&hcan2); 
+        uint8_t data[1] = {0};
+        uint32_t mailbox; //should this be equal to the value HAL_CAN_GetTxMailboxesFreeLevel(hcan2) ? 
```

Comment:
No it shouldn't be. This is needed by the transmit function. This actually indicates what mailbox the message was transmitted on. For this training, it's not needed, but the function still needs it for running purposes.